### PR TITLE
Update released versions via Gradle task

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: released-versions.json [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            "version": "1.4"
+        },
+        {
+            "buildTime": "20130125035146+0000",
+            "version": "1.4-rc-3"
+        },
+        {
+            "buildTime": "20130122030632+0000",
+            "version": "1.4-rc-2"
+        },
```
Comment (jjohannes) [inline released-versions.json]: ❌  should not be in this list I guess:
- `1.2-rc-1`
- `1.4-rc-2`
- `1.4-rc-3`

**Comment by jjohannes** [comment]
After this is merged, please also remove the line in the release process document stating _Update the compatibility baseline on master..._

https://docs.google.com/a/gradle.com/document/d/1h_Me4H4PfCRNUg5KLHht9zdSMNmMeOmQycvSyeBtyJ8/edit?disco=AAAABaCtIFc

File: released-versions.json (hunk 1 above)
Comment (wolfs) [reply released-versions.json]: Fixed!

**Comment by wolfs** [comment]
@jjohannes Yeah, doing the filtering by actually editing the `released-versions.json` file is a win. The filtering in `testedVersions` also is about to only test the first and last release of each major Gradle version in cross version tests to make them faster, right? I guess we still need to keep some of that in place.

**Comment by jjohannes** [comment]
@wolfs Yes we have one additional filtering in that method (`selection = true //Limit to first and last release of each major version`) which only selects a subset of all versions for the early build pipeline. We need to keep that one. I think reverting this commit would do it: https://github.com/gradle/gradle/commit/e4d398096cbb1f16ad292e2f166561c12a0f4a3a

**Pull request merged**